tests/conftest.py

```diff
@@ -453,6 +472,45 @@ def duthost(duthosts, request):
     return duthost
 
 
+@pytest.fixture(name="dpuhosts", scope="session")
+def fixture_dpuhosts(enhance_inventory, ansible_adhoc, tbinfo, request):
+    """
+    @summary: fixture to get DPU hosts defined in testbed.
+    @param ansible_adhoc: Fixture provided by the pytest-ansible package.
+        Source of the various device objects. It is
+        mandatory argument for the class constructors.
+    @param tbinfo: fixture provides information about testbed.
+    """
+    try:
+        host = DutHosts(ansible_adhoc, tbinfo, get_specified_dpus(request),
```

Comment:
Thank you for you quick response.
You mean when the smartswitch is up, the sonic-dpu-mgmt-traffic.sh will run automatically. Right?
If the script of sonic-dpu-mgmt-traffic.sh can save the config regradless of config reload or reboot, I think we can keep it as the scope as session.

Reply:
but I have question: how the sonic-dpu-mgmt-traffic.sh  to get the ssh ports information?
ansible_adhoc().options['inventory_manager'].get_host(<dpu hostname>).vars depends on the config of inventory, Right?

Reply:
I meant to call `sonic-dpu-mgmt-traffic.sh` at the start of every test session from inside of this fixture. If I understand the current implementation correctly, the user needs to manually call `sonic-dpu-mgmt-traffic.sh` on the NPU before starting the test session.

Also, I'm not sure if the configs changed by `sonic-dpu-mgmt-traffic.sh` will persist across a reload or reboot, it looks like mostly modifying iptables rules.

Reply:
If so, dpuhosts will depends on duthosts, we need to make sure to call duthosts before calling dpuhosts.
Additionally, currently sonic-dpu-mgmt-traffic.sh will not persist the config after reload or reboot. So, when the test does reload or reboot, we need to re-run the script again, which will make the implement of test more complex.
For now,  to persist the nat config, in our regression we run the following code on NPU during deploying the image. Maybe we can offer the other solution which is to add one pre-test to configure the NAT. What do you think?

> sudo su
> sudo  sed -i 's/#net.ipv4.ip_forward=1/net.ipv4.ip_forward=1/g' /etc/sysctl.conf
> sudo echo net.ipv4.conf.eth0.forwarding=1 >> /etc/sysctl.conf
> sudo sysctl -p
> sudo iptables -t nat -A POSTROUTING -s 169.254.200.0/24 -o eth0 -j MASQUERADE
> sudo iptables -t nat -A PREROUTING -i eth0 -p tcp --dport 5021 -j DNAT --to-destination 169.254.200.1:22
> sudo iptables -t nat -A PREROUTING -i eth0 -p tcp --dport 5022 -j DNAT --to-destination 169.254.200.2:22
> sudo iptables -t nat -A PREROUTING -i eth0 -p tcp --dport 5023 -j DNAT --to-destination 169.254.200.3:22
> sudo iptables -t nat -A PREROUTING -i eth0 -p tcp --dport 5024 -j DNAT --to-destination 169.254.200.4:22
> sudo iptables -t nat -L
> sudo iptables-save > /etc/iptables/rules.v4
>

Reply:
It sounds like a good idea, or possibly even doing it in a session-scoped fixture, but it might be better to address this in a separate PR. We can leave this as-is for now, could you add a short comment here about needing to call `sonic-dpu-mgmt-traffic.sh` before using the fixture?

Reply:
Done.